packages/squiggle-lang/src/fr/date.ts

```diff
@@ -2,145 +2,136 @@ import { REOther } from "../errors/messages.js";
 import {
   frDate,
+  frDict,
   frNumber,
   frString,
-  frTimeDuration,
+  frDuration,
 } from "../library/registry/frTypes.js";
-import { FnFactory } from "../library/registry/helpers.js";
 import {
-  type Duration,
-  duration,
-  date,
-  dateFromUnixS,
-  dateToUnixS,
-  dateFromString,
-} from "../utility/DateTime.js";
-import { vDate, vNumber, vTimeDuration } from "../value/index.js";
+  FnFactory,
+  makeNumericComparisons,
+} from "../library/registry/helpers.js";
+import { SDate } from "../utility/SDate.js";
+import { DateRangeDomain } from "../value/domain.js";
+import { vDate, vDomain, vNumber, vDuration } from "../value/index.js";
 
 const maker = new FnFactory({
   nameSpace: "Date",
   requiresNamespace: false,
 });
 
-const makeNumberToDurationFn = (name: string, fn: (v: number) => Duration) =>
-  maker.make({
-    name,
-    definitions: [makeDefinition([frNumber], ([t]) => vTimeDuration(fn(t)))],
-  });
-
 const makeYearFn = makeDefinition([frNumber], ([year]) => {
-  const result = date.makeFromYear(year);
+  const result = SDate.makeFromYear(year);
   if (!result.ok) {
     throw new REOther(result.value);
   }
   return vDate(result.value);
 });
 
-const makeDurationToNumberFn = (name: string, fn: (v: Duration) => number) =>
-  maker.make({
-    name,
-    definitions: [makeDefinition([frTimeDuration], ([t]) => vNumber(fn(t)))],
-  });
-
 export const library = [
+  ...makeNumericComparisons(
+    maker,
+    (d1, d2) => d1.smaller(d2),
+    (d1, d2) => d1.larger(d2),
+    (d1, d2) => d1.isEqual(d2),
+    frDate
+  ),
   maker.make({
     name: "make",
     requiresNamespace: true,
     examples: ['Date.make("2020-05-12")', "Date.make(2020, 5, 10)"],
+    output: "Date",
     definitions: [
       makeDefinition([frString], ([str]) => {
-        const result = dateFromString(str);
+        const result = SDate.fromString(str);
         if (!result.ok) {
           throw new REOther(result.value);
         }
         return vDate(result.value);
       }),
 
       makeDefinition([frNumber, frNumber, frNumber], ([yr, month, date]) => {
-        return vDate(new Date(yr, month - 1, date));
+        return vDate(SDate.fromYearMonthDay(yr, month, date));
       }),
     ],
   }),
   // same name as used in date-fns
   maker.make({
     name: "fromUnixTime",
+    examples: ["Date.fromUnixTime(1589222400)"],
     requiresNamespace: true,
+    output: "Date",
     definitions: [
       makeDefinition([frNumber], ([num]) => {
-        return vDate(dateFromUnixS(num));
+        return vDate(SDate.fromUnixS(num));
       }),
     ],
   }),
   maker.make({
     name: "toUnixTime",
+    examples: ["Date.toUnixTime(Date.make(2020, 5, 12))"],
     requiresNamespace: true,
+    output: "Number",
     definitions: [
       makeDefinition([frDate], ([date]) => {
-        return vNumber(dateToUnixS(date));
+        return vNumber(date.toUnixS());
       }),
     ],
   }),
   maker.fromDefinition("year", makeYearFn),
```

Comment:
`Date.make(2023,12,1) -> Date.year` also makes sense to me, maybe this should be `Date.fromYear`?

Reply:
Yea, I like that.